typescript/sdk/src/core/EvmCoreModule.ts

```diff
@@ -0,0 +1,267 @@
+  proxyFactoryFactories,
+} from '../deploy/contracts.js';
+import { HyperlaneIsmFactory } from '../ism/HyperlaneIsmFactory.js';
+import { MultiProvider } from '../providers/MultiProvider.js';
+import { EthersV5Transaction } from '../providers/ProviderType.js';
+import { ChainNameOrId } from '../types.js';
+
+import {
+  HyperlaneModule,
+  HyperlaneModuleArgs,
+} from './AbstractHyperlaneModule.js';
+import { EvmIcaModule } from './EvmIcaModule.js';
+import { HyperlaneCoreDeployer } from './HyperlaneCoreDeployer.js';
+import { CoreFactories } from './contracts.js';
+
+// Partial CoreConfig because will be filled once .deploy() is called
+type DeployedAdresses = Partial<HyperlaneAddresses<CoreFactories>> & {
+  testRecipient?: Address;
+  timelockController?: Address;
+  interchainAccountRouter?: Address;
+  interchainAccountIsm?: Address;
+  deployedIsmFactoryFactories: HyperlaneAddresses<ProxyFactoryFactories>;
+};
+
+export class EvmCoreModule extends HyperlaneModule<
+  ProtocolType.Ethereum,
+  CoreConfig,
+  DeployedAdresses
+> {
+  public readonly chainName: string;
+  protected logger = rootLogger.child({ module: 'EvmCoreModule' });
+  protected hyperlaneCoreDeployer: HyperlaneCoreDeployer;
+
+  protected constructor(
+    protected readonly multiProvider: MultiProvider,
+    args: HyperlaneModuleArgs<CoreConfig, DeployedAdresses>,
+  ) {
+    super(args);
+    this.chainName = this.multiProvider.getChainName(this.args.chain);
+
+    // Deploy IsmFactory to be used in CoreDeployer
+    const ismFactory = new HyperlaneIsmFactory(
+      attachContractsMap(
+        { [this.chainName]: this.args.addresses.deployedIsmFactoryFactories },
+        proxyFactoryFactories,
+      ),
+      multiProvider,
+    );
+
+    // Initalize Deployer
+    this.hyperlaneCoreDeployer = new HyperlaneCoreDeployer(
+      multiProvider,
+      ismFactory,
+    );
+  }
+
+  public async read(): Promise<CoreConfig> {
+    throw new Error('Method not implemented.');
+  }
+
+  public async update(_config: CoreConfig): Promise<EthersV5Transaction[]> {
+    throw new Error('Method not implemented.');
+  }
+
+  /**
+   * Deploys the Core contracts.
+   *
+   * @remark Most of the contract owners are the deployers with some being the Proxy Admin.
+   *
+   * @param chain - The chain name or ID to deploy the Hyperlane contracts on.
+   * @param config - The derived core configuration for the deployment.
+   * @param multiProvider - The multi-provider instance to use for the deployment.
+   * @returns The created EvmCoreModule instance.
+   */
+  public static async create({
+    chain,
+    config,
+    multiProvider,
+  }: {
+    chain: ChainNameOrId;
+    config: CoreConfig;
+    multiProvider: MultiProvider;
+  }): Promise<EvmCoreModule> {
+    const chainName = multiProvider.getChainName(chain);
+    // Deploy Ism Factories
+    const ismFactoriesFactory = await EvmCoreModule.deployIsmFactories(
+      chainName,
+      config,
+      multiProvider,
+    );
```

Comment:
slightly surprised this is outside of the `deploy` itself

Reply:
fixed. moved into deploy